flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcRowDataLookupFunction.java

```diff
@@ -92,9 +103,22 @@ public JdbcRowDataLookupFunction(
                                 })
                         .toArray(DataType[]::new);
         this.maxRetryTimes = maxRetryTimes;
-        this.query =
+
+        final String baseSelectStatement =
                 options.getDialect()
                         .getSelectFromStatement(options.getTableName(), fieldNames, keyNames);
+        if (conditions == null || conditions.length == 0) {
+            this.query = baseSelectStatement;
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Issuing look up select {}", this.query);
+            }
+        } else {
+            this.query = baseSelectStatement + " AND " + String.join(" AND ", conditions);
```

Comment:
What if the `baseSelectStatement` does not contain a `WHERE` clause?

Reply:
[https://github.com/apache/flink-connector-jdbc/blob/e3dd84160cd665ae17672da8b6e742e61a72a32d/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/AbstractDialect.java#L198](https://github.com/apache/flink-connector-jdbc/blob/e3dd84160cd665ae17672da8b6e742e61a72a32d/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/AbstractDialect.java#L198) puts in a WHERE clause if there are conditions and this else is only when there are conditions.